cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/ActiveTimController.java

```diff
@@ -47,1550 +48,1208 @@
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(activeTims);
+        }
+        return ResponseEntity.ok(activeTims);
+    }
+
+    @RequestMapping(value = "/expired", method = RequestMethod.GET)
+    public ResponseEntity<List<ActiveTim>> GetExpiredActiveTims() {
+        List<ActiveTim> activeTims = new ArrayList<ActiveTim>();
+
+        String selectStatement = "select * from ACTIVE_TIM";
+        selectStatement += " WHERE TIM_END <= (NOW() AT TIME ZONE 'UTC')";
+
+        try (Connection connection = dbInteractions.getConnectionPool();
+             Statement statement = connection.createStatement();
+             ResultSet rs = statement.executeQuery(selectStatement)) {
+            activeTims = getActiveTimFromRS(rs, false);
+        } catch (SQLException e) {
+            log.error("Error getting expired active tims", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(activeTims);
+        }
+
+        return ResponseEntity.ok(activeTims);
+    }
+
+    @RequestMapping(value = "/indices-rsu/{rsuTarget}", method = RequestMethod.GET)
+    public ResponseEntity<List<Integer>> GetActiveTimIndicesByRsu(@PathVariable String rsuTarget) {
+
+        List<Integer> indices = new ArrayList<>();
+
+        String selectStatement = "select tim_rsu.rsu_index from active_tim";
+        selectStatement += " inner join tim on active_tim.tim_id = tim.tim_id";
+        selectStatement += " inner join tim_rsu on tim_rsu.tim_id = tim.tim_id";
+        selectStatement += " inner join rsu on rsu.rsu_id = tim_rsu.rsu_id";
+        selectStatement += " inner join rsu_view on rsu.deviceid = rsu_view.deviceid";
+        selectStatement += " where rsu_view.ipv4_address = '" + rsuTarget + "'";
+
+        try (Connection connection = dbInteractions.getConnectionPool();
+             Statement statement = connection.createStatement();
+             ResultSet rs = statement.executeQuery(selectStatement)) {
+            // convert to ActiveTim object
+            while (rs.next()) {
+                indices.add(rs.getInt("RSU_INDEX"));
+            }
+        } catch (SQLException e) {
+            log.error("Error getting active tim indices by rsu", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(indices);
+        }
+
+        return ResponseEntity.ok(indices);
+    }
+
+    @RequestMapping(value = {"/client-id-direction/{clientId}/{timTypeId}",
+        "/client-id-direction/{clientId}/{timTypeId}/{direction}"}, method = RequestMethod.GET)
+    public ResponseEntity<List<ActiveTim>> GetActiveTimsByClientIdDirection(
+        @PathVariable String clientId, @PathVariable Long timTypeId,
+        @PathVariable(required = false) String direction) {
+        List<ActiveTim> activeTims = new ArrayList<>();
+
+        // There may be multiple TIMs grouped together by client_id. ex. CLIENTID_1,
+        // CLIENTID_2

```

Comment:
Note: This comment may no longer be accurate since we transitioned to multi-region TIMs instead of grouping them this way. It would be beneficial to review the system for any remaining instances of this expectation and update them as needed.

Reply:
do we need to add this to the backlog?

Reply:
Added! https://dev.azure.com/trihydro/CV/_workitems/edit/14835